### PR TITLE
LCD support, misc. other fixes

### LINKED ISSUE #15: Add support for LiquidCrystal
LiquidCrystal is the built in LCD lib that ships with Arduino, see also: http://www.arduino.cc/en/Reference/LiquidCrystal

I'm not going to pretend that this will be a simple "drop-in" addition, it will definitely require a new handler in du.ino, but I can promise that it will be totally awesome!

### PR DESCRIPTION
**Pull request by natevw**
Would love to see these changes integrated into trunk. On my branch I have:
- updated the documentation for code as I found it (message types, Ping plugin)
- fixed a couple off-by-one assignments I noticed in du.ino
- **added LCD support** by porting LiquidCrystal to a JS plugin (fixes #15, no protocol changes needed)
- documented LCD support, updated package.js

Big thanks to [&yet](http://andyet.com) for sponsoring this open source work!

### CODE REVIEW

**Comment by ecto** [comment]
Wow, this is awesome! I'm pretty impressed that it was only a small change to du.ino. I'm traveling right now, so I'll take a deeper look at this when I get back. Really appreciate your effort on this.

**Comment by adamavenir** [comment]
@ecto is this likely to get merged in?

File: lib/lcd.js [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+LCD_MOVERIGHT = 0x04;
+LCD_MOVELEFT = 0x00;
+
+// flags for function set
+LCD_8BITMODE = 0x10;
+LCD_4BITMODE = 0x00;
+LCD_2LINE = 0x08;
+LCD_1LINE = 0x00;
+LCD_5x10DOTS = 0x04;
+LCD_5x8DOTS = 0x00;
```
Comment (ecto) [inline lib/lcd.js:40]: Why are these globals?

**Comment by ecto** [comment]
Hey, sorry for the horrifically long wait. I was traveling again, moving, working, etc. and forgot about this. Merging now :)

**Comment by ecto** [comment]
Also hell yeah. Great work

File: lib/lcd.js (hunk 1 above)
Comment (natevw) [inline lib/lcd.js:40]: I can't really recall if there was a particular reason or not, but unless you're doing something fancy when loading subcomponents, they are module-scoped and shouldn't affect (or be affected by) anyone else's code.

**Comment by natevw** [comment]
Thanks, glad you like it! (And thanks again @andyet / @adambrault for sponsoring the project :-)

**Pull request merged**